### ISSUE TITLE
ES Modules syntax support

### ISSUE THREAD

**Issue by ginpei** [mozilla/addons-linter#2040]
### Describe the problem and steps to reproduce it:
(Please include as many details as possible.)

Since latest Firefox has supported ES Modules, it would be great to use them even if you are working on addons. Would you add ES Modules syntax support?

### What happened?

I created an addon using ES Modules, tried to publish on the store but it was rejected because of lint error.

```console
$ npx addons-linter extension
Validation Summary:

errors          3
notices         1
warnings        1

ERRORS:

Code              Message        Description                      File               Line   Column
JS_SYNTAX_ERROR   JavaScript     There is a JavaScript syntax     background/Assi…   2      1
                  syntax error   error in your code; validation   stantUI.js
                                 cannot continue on this file.
JS_SYNTAX_ERROR   JavaScript     There is a JavaScript syntax     background/Back…   1      1
                  syntax error   error in your code; validation   groundControlle…
                                 cannot continue on this file.    r.js
JS_SYNTAX_ERROR   JavaScript     There is a JavaScript syntax     background/main…   1      1
                  syntax error   error in your code; validation   .js
                                 cannot continue on this file.
...
```

One of the scripts:

```js
// eslint-disable-next-line no-unused-vars
export default class AssistantUI {
...
```

And the other one:

```js
import AssistantUI from './AssistantUI.js';
...
```

### What did you expect to happen?

Code using ES Modules is accepted.

### Anything else we should know?
(Please include screenshots and any relevant files.)

**Comment by wagnerand** [comment]
Duplicate of #1775.
